### ISSUE TITLE
Step debugging tries to evaluate infinite seq and hangs

### ISSUE THREAD

**Issue by yaronthurm** [BetterThanTomorrow/calva#631]
When using the "Instrument Top Level Form for debugging" and stepping into individual forms, if there is an infinite seq as one of the steps, the debugger will try to evaluate it and will hang.
Take this code as an example:
<img width="252" alt="Screen Shot 2020-04-30 at 19 52 49" src="https://user-images.githubusercontent.com/3634325/80737474-3e5f4e80-8b1c-11ea-8c7e-64467ae812c0.png">

When starting debugging, the debugger first evaluates "a"
<img width="389" alt="Screen Shot 2020-04-30 at 19 46 35" src="https://user-images.githubusercontent.com/3634325/80736847-600c0600-8b1b-11ea-9001-6388464ea098.png">

When trying to "Step Over"/"Step Into" the debugger will start to evaluate "(range)" and will hang.

**Comment by bpringe** [comment]
Thanks for reporting. We delegate to cider-nrepl for the evaluation of next form, etc., but we may be able to fix this issue on our end with print-length settings.

**Comment by bpringe** [comment]
So, short of this being "fixed" upstream in cider-nrepl to handle special but common cases like `(range)`, this can be fixed by setting `*print-length*` before you debug. Just like when not debugging, if you evaluate `(range)` in a repl it will lock up if there is no `*print-length*` set (or if it's not set to a reasonable number).

```
(set! *print-length* 3)
(range)
```

Evaluating the above lines will result in no lockup / out-of-memory error. @PEZ, we have a couple options here:
1. We do nothing but document this in the debugger (setting *print-length* for debugging with infinite sequences)
2. We temporarily set `*print-length*` to some low number when debugging starts - saving its previous value, and setting it back at the end of the debugging session. The user can still change this during debugging if desires via `set!` - and also we would document this.

I think `2` would be wise because this will pop up again with other users in the future.

**Comment by bpringe** [comment]
@PEZ Ok, so the separate session idea didn't work as well as I thought. So we're back to the above options.

The problem with `2` is that debugging is started by evaluating an instrumented form, and Calva doesn't have any indication that cider-nrepl has started debugging until the first `need-debug-response` nrepl message. So with the below form, evaluated with `ctrl+alt+c i`,

```
(let [a [1 2 3]
      b (map + (range) a)]
  b)
````

the first thing that cider-nrepl evaluates in debug mode is `(range)` - so by the time we know we're debugging, the repl is hung, assuming `*print-length*` is still `nil`, the default.

So given all of that info, it seems best to leave setting `*print-length*` up to the user, and we can document it in the debugger docs. I think defaulting that var to some value would annoy more people than it would help. Thoughts?

**Comment by bpringe** [comment]
I've just realized that along with leaving the `*print-length*` value up to the user in this context, we also need to make sure we set it to `false` when getting the lint analysis for debug decorations so we get full results, and then set it back to what it was afterward. See #669.

I'll update the debug docs to explain setting `*print-length*` to avoid this hanging on infinite seqs during debugging, and then we can close this issue.
